awscli/customizations/wizard/ui/prompt.py

```diff
@@ -0,0 +1,120 @@
+# language governing permissions and limitations under the License.
+from prompt_toolkit.application import get_app
+from prompt_toolkit.buffer import Buffer
+from prompt_toolkit.document import Document
+from prompt_toolkit.key_binding import KeyBindings
+from prompt_toolkit.layout.containers import (
+    Window, VSplit, Dimension
+)
+from prompt_toolkit.layout.controls import BufferControl
+
+
+class WizardPrompt:
+    def __init__(self, value_name, value_definition):
+        self._value_name = value_name
+        self._value_definition = value_definition
+        self.container = self._get_container()
+
+    def _get_container(self):
+        answer_container = WizardPromptAnswer(self._value_name)
+        answer_buffer = answer_container.buffer
+        return VSplit(
+            [
+                WizardPromptDescription(
+                    answer_buffer,
+                    self._value_definition['description']
+                ),
+                answer_container
+            ],
+            key_bindings=self._get_key_bindings(answer_buffer)
+        )
+
+    def _get_key_bindings(self, answer_buffer):
+        kb = KeyBindings()
+
+        def submit_current_answer(event):
+            event.app.values[self._value_name] = answer_buffer.text
+
+        @kb.add('tab')
+        @kb.add('enter')
+        def next_prompt(event):
+            submit_current_answer(event)
+            event.app.layout.focus_next()
+
+        @kb.add('s-tab')
+        def last_prompt(event):
+            submit_current_answer(event)
+            event.app.layout.focus_last()
+
+        return kb
+
+    def __pt_container__(self):
+        return self.container
+
+
+class WizardPromptDescription:
+    def __init__(self, answer_buffer, value_description):
+        self._answer_buffer = answer_buffer
+        self._value_description = value_description
+        self.container = self._get_container()
+
+    def _get_container(self):
+        content = f'{self._value_description}:'
+        buffer = Buffer(
+            document=Document(content),
+            read_only=True
+        )
+        return Window(
+            content=BufferControl(
+                buffer=buffer, focusable=False
+            ),
+            style=self._get_style,
+            width=Dimension.exact(len(content) + 1),
+            dont_extend_height=True,
+        )
+
+    def _get_style(self):
+        if get_app().layout.has_focus(self._answer_buffer):
+            return 'class:wizard.prompt.description.current'
+        else:
+            return 'class:wizard.prompt.description'
+
+    def __pt_container__(self):
+        return self.container
+
+
+class WizardPromptAnswer:
+    def __init__(self, value_name):
+        self._value_name = value_name
+        self.buffer = self._get_answer_buffer()
+        self.container = self._get_answer_container()
+
+    def _get_answer_buffer(self):
+        return Buffer(name=f'{self._value_name}_answer_buffer')
```

Comment:
What the reason of giving name to this Buffer if we never use this name?

I was wondering if it makes sense to name the buffer just `f'{self._value_name}` and make keybindings global instead of injecting them into each `WizardPrompt` 

Something like

```python

def submit_current_answer(event):
    current_buffer = event.app.current_buffer.name
    event.app.values[current_buffer.name] = .current_buffer.text

@kb.add('tab')
@kb.add('enter')
def next_prompt(event):
    submit_current_answer(event)
    event.app.layout.focus_next()

@kb.add('s-tab')
def last_prompt(event):
    submit_current_answer(event)
    event.app.layout.focus_last()
```

Reply:
I think I originally wanted to use the name but it was something that got missed after trying a couple of different approaches. Honestly the key tabbing logic is going to be going away in the next PR, specifically I'm going to introduce an abstraction that walks though the wizard prompts telling us what the current prompt is and what the next prompt will be. I do like the idea of making the keybindings global now though so I'll introduce a `keybindings` module to encompass the current logic and build off of it.